### PR TITLE
feat(dashboard): added collapsable sidebar with routes and subroutes

### PR DESCRIPTION
**Pull request by mostafakamar2308**
The side bar is adaptable for both routes and subroutes,

@neuodev  @moalidv 
> *the schemas for how to write new routes and subroutes for the dashboard are in [here](https://github.com/litespace-org/litespace/blob/09ea7c96e557a0b9d16145b2b80feb4e132084b4/apps/dashboard/src/types/navbar.ts#L13)*

### CODE REVIEW

**Comment by neuodev** [comment]
🚀 Nova: https://nova-1kmdwyul5-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-rfjz1944y-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-oybx6kgwi-litespace-cores-projects.vercel.app

File: apps/dashboard/src/components/common/SideBarAccordion.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+              <ChevronDownIcon />
+            </div>
+          </div>
+          <div
+            data-open={item.id === open}
+            className={cn(
+              "tw-text-foreground-light",
+              "data-[open=true]:tw-border-b data-[open=true]:tw-border-border-strong",
+              "tw-grid tw-grid-rows-[0fr] data-[open=true]:tw-grid-rows-[1fr]",
+              "tw-transition-[grid-template-rows] tw-duration-200 tw-ease-in tw-px-4 data-[open=true]:tw-py-2"
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBarAccordion.tsx:53]: Are you sure that this the right way to do it? Because the `tw-` prefix doesn't work on the dashboard. @mostafakamar2308

Comment (mostafakamar2308) [reply apps/dashboard/src/components/common/SideBarAccordion.tsx]: @neuodev 
I didn't see any problems but I removed it to make the style consistent with the dashboard itself

**Comment by neuodev** [comment]
🚀 Nova: https://nova-cj971mvel-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-3r8q5foqz-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-fgdzig1om-litespace-cores-projects.vercel.app

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          onClick={toggleMenu}
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
+        >
+          <Sidebar className="w-8 h-8" />
+        </Button>
+        <Button type={ButtonType.Text} onClick={toggleMenu}>
+          {profile?.image ? (
+            <img src={profile.image} />
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:67]: Make sure to add something styling to the image. As I think it will not look good.

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                    onChange={toggleTheme}
+                  />
+                  <Moon />
+                </div>
+                <Button
+                  onClick={logout}
+                  className="w-full"
+                  type={ButtonType.Error}
+                >
+                  Log out
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:129]: Remember to always use `intl` to render text and not use plain text.

File: apps/dashboard/src/components/common/SideBarAccordion.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useState } from "react";
+import { ChevronDownIcon } from "@radix-ui/react-icons";
+import cn from "classnames";
+import { NavAccordionItem } from "@/types/navbar";
+import SideBarItem from "./SideBarItem";
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBarAccordion.tsx:5]: Please use absolute paths.

File: apps/dashboard/src/pages/Root.tsx [hunk 5]
```diff
@@ -22,7 +49,14 @@ const Root: React.FC = () => {
   }, [location.pathname, navigate, profile]);
 
   return (
-    <main className="min-h-screen text-foreground flex overflow-y-hidden">
+    <main
+      className={`grid ${
+        !profile ? "" : "grid-cols-[5%,95%]"
+      } min-h-screen overflow-y-hidden text-foreground`}
```
Comment (neuodev) [inline apps/dashboard/src/pages/Root.tsx:55]: Please use the `classnames` library to add conditional class names. You can look into other files through out the codebase to get a sense on how to use it.

File: apps/dashboard/src/pages/Root.tsx [hunk 6]
```diff
@@ -22,7 +49,14 @@ const Root: React.FC = () => {
...
+    >
+      <div className={`${!profile ? "hidden" : ""}`}>
```
Comment (neuodev) [inline apps/dashboard/src/pages/Root.tsx:57]: Please use `classnames`

File: packages/luna/src/locales/ar-eg.json [hunk 7]
```diff
@@ -233,6 +233,10 @@
   "invoices.process.submit.approveUpdateRequest": "قبول تحديث بيانات طلب السحب",
   "invoices.process.success": "تم تحديث بيانات طلب السحب بنجاح",
   "invoices.process.error": "لديك مشكلة في تحديث بيانات طلب السحب",
+  "dashboard.navbar.users": "المستخدمون",
+  "dashboard.navbar.invoices": "إيصالات الدفع",
+  "dashboard.navbar.media": "الوسائط المتعددة",
+  "dashboard.navbar.plans": "خطط الدفع",
```
Comment (neuodev) [inline packages/luna/src/locales/ar-eg.json:239]:
```suggestion
  "dashboard.navbar.users": "المستخدمين",
  "dashboard.navbar.invoices": "طلبات السحب",
  "dashboard.navbar.media": "صور وفديوهات المعلمين",
  "dashboard.navbar.plans": "الخطط",
```

**Review by neuodev — CHANGES_REQUESTED** [review]
Great work @mostafakamar2308. The task is well executed. Please take a look at my minor comments.

**Comment by neuodev** [comment]
🚀 Nova: https://nova-l4ipbgams-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-5whti0rrm-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-mdjap7gg4-litespace-cores-projects.vercel.app

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+import { SideBarAccordion } from "@/components/common/SideBarAccordion";
+import { NavAccordionItem, NavOption } from "@/types/navbar";
+import SideBarItem from "@/components/common/SideBarItem";
+
+const framerSidebarBackground = {
+  initial: { x: "30rem", y: "0rem" },
+  animate: { x: "0rem", y: "0rem" },
+  exit: { x: "20rem", y: "0rem" },
+  transition: { duration: 0.2 },
+};
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:29]: It is a good practice to always assign types to your variables. 
```suggestion
} as const;
```

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
+        >
+          <Sidebar className="w-8 h-8" />
+        </Button>
+        <Button type={ButtonType.Text} onClick={toggleMenu}>
+          {profile?.image ? (
+            <div className="overflow-hidden border-2 border-white rounded-full">
+              <img src={profile?.image} className="w-10 h-10 rounded-full" />
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:70]:
```suggestion
              <img src={profile.image} className="w-10 h-10 rounded-full" />
```

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+                  {profile?.image ? (
+                    <div className="overflow-hidden border-2 border-white rounded-full">
+                      <img
+                        src={profile.image}
+                        className="w-10 h-10 rounded-full"
+                      />
+                    </div>
+                  ) : (
+                    <User className="w-8 h-8" />
+                  )}
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:105]: I can see this pattern repeated twice. You can consider moving it into a shared component and then reuse it.
```suggestion
                  <UserImage />
```

**Review by neuodev — CHANGES_REQUESTED** [review]
image

Always give attention to edge cases. For example, when the test is so long. 

Please truncate the test when it is so long.

**Comment by neuodev** [comment]
image

- Let's try to put these side by side (`flex-row`)
- Also, make the button smaller by use `ButtonSize.Small`

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+          onClick={toggleMenu}
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
+        >
+          <Sidebar className="w-8 h-8" />
+        </Button>
+        <Button type={ButtonType.Text} onClick={toggleMenu}>
+          {profile?.image ? (
+            <div className="overflow-hidden border-2 border-white rounded-full">
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:69]: image

The white border doesn't fit the design. Please choose one of these borders to fit you needs. 

https://github.com/litespace-org/litespace/blob/12d615ad527b0d26a0312907747a4f0edb607229/packages/luna/tailwind.config.js#L90-L103

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+        ref={button}
+        className="fixed flex flex-col items-center justify-between h-screen p-2 border-l"
+      >
+        <Button
+          onClick={toggleMenu}
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
+        >
+          <Sidebar className="w-8 h-8" />
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:65]:
```suggestion
          <Sidebar className="w-8 h-8 text-foreground" />
```

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+    toggle: toggleMenu,
+    open,
+    ref,
+  } = useClosableRef<HTMLUListElement>(button.current);
+
+  const logout = useCallback(() => {
+    removeToken();
+    dispatch(resetUserProfile());
+    navigate(Route.Login);
+  }, [dispatch]);
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:51]: You are missing a dependency in your `useCallback`. To avoid this you can enable the `eslint` extension.

File: apps/dashboard/src/components/common/SideBar.tsx [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+    <nav className="relative">
+      <div
+        ref={button}
+        className="fixed flex flex-col items-center justify-between h-screen p-2 border-l"
+      >
+        <Button
+          onClick={toggleMenu}
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
```
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:63]: The padding didn't take effect because you already have the size set to small. Remove the size for the padding to take effect.

File: apps/dashboard/src/components/common/SideBar.tsx (hunk 12 above)
Comment (neuodev) [inline apps/dashboard/src/components/common/SideBar.tsx:65]: Let's make our icons smaller. Please make all icons `w-6 h-6`. 
```suggestion
          <Sidebar className="w-6 h-6" />
```

**Review by neuodev — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/components/common/SideBar.tsx (hunk 14 above)
Comment (mostafakamar2308) [reply apps/dashboard/src/components/common/SideBar.tsx:63]: Nice 🚀

**Comment by neuodev** [comment]
🚀 Nova: https://nova-7kgqgq35z-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-difavsyt7-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-7v4ucl7rg-litespace-cores-projects.vercel.app

**Comment by neuodev** [comment]
🚀 Nova: https://nova-d54bjjvjk-litespace-cores-projects.vercel.app
🚀 Luna: https://luna-9xo4fdvot-litespace-cores-projects.vercel.app
🚀 Nebula: https://nebula-1rabuiz77-litespace-cores-projects.vercel.app
🚀 Dashboard: https://dashboard-odf5m81df-litespace-cores-projects.vercel.app

**Review by neuodev — APPROVED** [review]
